contracts/HashRegistrarSimplified.sol

```diff
@@ -431,13 +431,13 @@ contract Registrar {
      * @return The hash of the bid values
      */
     function shaBid(bytes32 hash, address owner, uint value, bytes32 salt) public pure returns (bytes32) {
-        return keccak256(hash, owner, value, salt);
+        return keccak256(abi.encodePacked(hash, owner, value, salt));
```

Comment:
Pretty sure this should be `abi.encode`; `encodePacked` will pack the address as 160 bits instead of 256.

Reply:
Can you try a test with an address type, and no variable length (string, bytes) types?

Reply:
pragma solidity ^0.4.24;

    contract echoEncoding{
        event echo(bytes32 echoHash);
        function echoEncoding(){
            address addr1 =  0x314159265dd8dbb310642f98f50c066173c1259b;
            emit echo(keccak256(addr1));
            emit echo(keccak256(abi.encode(addr1)));
            emit echo(keccak256(abi.encodePacked(addr1)));
       }
    } 

Output


    "echoHash": "0x47c21aaccf7858a0bbed6eb11df8b02b87bb4281c9b605ca5302417556015435"
    "echoHash": "0x40a1e975a27fc25470c2a4433cdddd8a359d3114fa0628c5b96f7c954bce754f"
    "echoHash": "0x47c21aaccf7858a0bbed6eb11df8b02b87bb4281c9b605ca5302417556015435"